mail/models.py

```diff
@@ -0,0 +1,12 @@
+""" Models for mail app """
+from django.db import models
+
+from open_discussions.models import TimestampedModel
+
+
+class BlockedEmailRegex(TimestampedModel):
```

Comment:
I'm in the process of extracting the `authentication` app out into a reusable django app, and I think this model should live in the `authentication` app instead so I can include it in that effort and all our apps can benefit. I'd probably also have the check for the blocked email happen in a pipeline action, but I'm ok with it living in ` mail/verification_api.py` for the sake of getting this out sooner. Moving the model later will be more difficult though.